### ISSUE TITLE
earlyexittest fails on OS X with Qt version 5.4 and above

### ISSUE THREAD

**Issue by krf** [KDAB/GammaRay#318]
Took me a while to figure out what's causing this (first assumed a problem with `QProcess::waitForStarted`...).

The real cause of the problem lies in injectorfactory.cpp:
```
AbstractInjector::Ptr defaultInjectorForLaunch(const ProbeABI &abi)
{
#if defined(Q_OS_MAC)
    if (abi.majorQtVersion() >= 5 && abi.minorQtVersion() >= 4)
        return createInjector(QStringLiteral("preload"));
    return findFirstWorkingInjector(QStringList() << QStringLiteral("lldb")
                                                  << QStringLiteral("gdb"));
#elif defined(Q_OS_UNIX)
    Q_UNUSED(abi);
    return createInjector(QStringLiteral("preload"));
#else
```

=> For OS X & Qt <5.4: We use lldb by default as injector.

=> For those configurations `EarlyExitTest::testNonExistingTarget` will use lldb as injector.

The assertion `QVERIFY(!launcher.start());` fails in this case, because the debugger indeed starts fine!

**Comment by krf** [comment]
Relevant output from CI:

```
29: Test command: /Users/build/Buildbot/gammaray-qt53-osx/build/bin/earlyexittest
29: Test timeout computed to be: 9.99988e+06
29: ********* Start testing of EarlyExitTest *********
29: Config: Using QtTest library 5.3.2, Qt 5.3.2
29: PASS   : EarlyExitTest::initTestCase()
29: FAIL!  : EarlyExitTest::testNonExistingTarget() '!launcher.start()' returned FALSE. ()
29:    Loc: [/Users/build/Buildbot/gammaray-qt53-osx/source/tests/earlyexittest.cpp(65)]
29: PASS   : EarlyExitTest::test()
29: Injector error: Process stopped.
29: PASS   : EarlyExitTest::testStop(default)
29: Injector error: Process stopped.
29: PASS   : EarlyExitTest::testStop(lldb)
29: PASS   : EarlyExitTest::cleanupTestCase()
29: Totals: 5 passed, 1 failed, 0 skipped
29: ********* Finished testing of EarlyExitTest *********
29/29 Test #29: earlyexittest ....................***Failed    7.93 sec
```

**Comment by krf** [comment]
Fixed by: 766e0fc0af2d4cedd53ef1b8d547629a4eaf5ff5
